core/src/main/scala/akka/persistence/r2dbc/snapshot/SnapshotDao.scala

```diff
@@ -132,24 +132,19 @@ private[r2dbc] final class SnapshotDao(settings: R2dbcSettings, connectionFactor
 
     sql"""
       DELETE FROM $snapshotTable
-      WHERE slice = ? AND entity_type = ? AND persistence_id = ?
+      WHERE persistence_id = ?
```

Comment:
I get removing the slice, but doesn't removing the `entity_type` make it so you can only ever store a single type of entity with the same persistence id, or do we expect that it will always be in the pid (what if someone is using the journal without sharding for example)? Worth a mention in the docs at least?

Reply:
The `persistenceId` must be unique across different entity types. That's historical reasons. Entity type was added much later. That's why we create the persistenceId string with the `PersistenceId.apply(entityTypeHint, entityId)` and it becomes `entityTypeHint|entityId`.

It's the same in all other persistence plugins.

The code here to retrieve the entityType was to extract it from the persistenceId.

Reply:
I see, I didn't know, thanks for the clarification!